src/Components/Contributors/ContributorSearchPanel.tsx

```diff
@@ -1,40 +1,218 @@
-import React, { FC } from 'react';
-import { Button, Card, CardContent, Collapse } from '@material-ui/core';
-import { ContributorType } from '../../types/ContributorTypes';
+import React, { FC, useEffect, useState } from 'react';
+import { Button, Card, CardContent, CircularProgress, Collapse, Typography } from '@material-ui/core';
+import { ContributorType, ContributorWrapper } from '../../types/ContributorTypes';
 import ContributorSearchResultItem from './ContributorSearchResultItem';
+import { getPersonDetailById, searchPersonDetailByName } from '../../api/contributorApi';
+import { Affiliation } from '../../types/InstitutionTypes';
+import { getAffiliationDetails } from '../../utils/contributorUtils';
+import { handlePotentialExpiredSession } from '../../api/api';
+import styled from 'styled-components';
+import { Colors } from '../../assets/styles/StyleConstants';
+import clone from 'just-clone';
+
+const StyledResultTypography = styled(Typography)`
+  &.MuiTypography-root {
+    margin-top: 1rem;
+    margin-bottom: 1rem;
+  }
+  color: ${Colors.Text.OPAQUE_87_BLACK};
+`;
+
+export interface AddAffiliationError extends Error {
+  institutionId: string;
+}
 
 interface ContributorSearchPanelProps {
-  searchResult: any[];
-  collapsed: boolean;
-  handleChoose: (author: any) => void;
-  handleAbort: () => void;
+  resultListIndex: number;
+  contributorData: ContributorWrapper;
+  updateContributor: (contributorData: ContributorWrapper, rowIndex: number) => void;
 }
 
 const ContributorSearchPanel: FC<ContributorSearchPanelProps> = ({
-  searchResult,
-  collapsed,
-  handleChoose,
-  handleAbort,
+  resultListIndex,
+  contributorData,
+  updateContributor,
 }) => {
-  return searchResult.length > 0 ? (
-    <Collapse in={collapsed}>
-      <Card>
-        <CardContent>
-          {searchResult.map((author: ContributorType) => (
-            <ContributorSearchResultItem
-              key={author.cristin_person_id}
-              contributor={author}
-              handleChoose={handleChoose}
-            />
-          ))}
-        </CardContent>
-        <Button color="primary" onClick={handleAbort}>
-          Lukk
-        </Button>
-      </Card>
-    </Collapse>
-  ) : (
-    <></>
+  const [searchResults, setSearchResults] = useState<ContributorType[]>([]);
+  const [searching, setSearching] = useState(false);
+  const [searchError, setSearchError] = useState<Error | undefined>();
+  const [openContributorSearchPanel, setOpenContributorSearchPanel] = useState(false);
+  const [addAffiliationSuccessful, setAddAffiliationSuccessful] = useState<string | undefined>(undefined);
```

Comment:
skjønte jeg ikke. Er det en verdi som gir status for sist lagt til instutusjon  ?